design-docs/dependency-management-projectpriority.md

```diff
@@ -0,0 +1,36 @@
+# Use case
```

Comment:
I'd pluralize this section and try to describe all the "Use Cases" for dependency resolution which will be affected by this change.

Obviously, one is your original use case. "I want to prefer project dependencies over external artifact dependencies even when the version number of the external dependencies is greater."

Another use case we have to consider is "I want to be able to explicitly prefer a different dependency to enable a project depending on an older version of itself in one of the circumstances where we allow circular dependencies."

Maybe include examples. Be sure to explain in the design how you are going to solve this problem. We may need more discussion on this topic. I am still nervous about the idea of using a flag to change the behavior. It seems like it would be better to add a new way to change the behavior explicitly in the DSL.